### ISSUE TITLE
address TODOs

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/dot#95]
Several TODOs and dead code in Quaternion.js with no associated GitHub issue:

```js
6 * TODO: convert from JME-style parametrization into classical mathematical description?
```

```js
96    // TODO: note why this is the case? product noted everywhere is the other one mentioned!
      // mathematica-style
//        return new Quaternion(
//                this.x * quat.x - this.y * quat.y - this.z * quat.z - this.w * quat.w,
//                this.x * quat.y + this.y * quat.x + this.z * quat.w - this.w * quat.z,
//                this.x * quat.z - this.y * quat.w + this.z * quat.x + this.w * quat.y,
//                this.x * quat.w + this.y * quat.z - this.z * quat.y + this.w * quat.x
//        );
```

```js
138      // TODO: optimization?
```

**Comment by pixelzoom** [comment]
This is apparently a pervasive problem in dot.  Moving to https://github.com/phetsims/dot/issues/96.

**Comment by phet-dev** [comment]
Reopening because there is a TODO marked for this issue.

**Comment by marlitas** [comment]
This never got closed. Ironically, because of a TODO issue. I committed a fix that I accidentally linked to #96. Closing.
